test/basic/ecs_compute_update/ecs_compute_update_test.go

```diff
@@ -0,0 +1,193 @@
+	defer cleanupAndAssert(t, generateMutableTerraformOptions())
+	terraform.InitAndPlan(t, generateMutableTerraformOptions())
+	terraform.Apply(t, generateMutableTerraformOptions())
+
+	c := client.NewClient(
+		true,
+		os.Getenv("APTIBLE_HOST"),
+		os.Getenv("APTIBLE_TOKEN"),
+	)
+	ctx := context.Background()
+
+	vpcId := terraform.Output(t, terraformOptions, "vpc_id")
+	vpcAsset, vpcAws, err := getAptibleAndAWSVPCs(t, ctx, c, vpcId, "testecs-compute-vpc")
+	assert.Nil(t, err)
+
+	ecsComputeId := terraform.Output(t, terraformOptions, "ecs_compute_id")
+	ecsComputeAsset, ecsClusterAws, ecsServiceAws, err := getAptibleAndAWSECSServiceAndCluster(t, ctx, c, ecsComputeId, "ecs-compute-test-compute-cluster", "ecs-compute-test")
+	assert.Nil(t, err)
+	assertCommonValues(t, vpcId, ecsComputeId, vpcAsset, vpcAws, ecsComputeAsset, ecsClusterAws, ecsServiceAws)
+
+	type atomicChange struct {
+		key      string
+		updated  interface{}
+		original interface{}
+	}
+	for _, changeSet := range [][]atomicChange{
+		// change container name
+		{{
+			key:      "compute_name",
+			updated:  "ecs-compute-uname-only-test",
+			original: "ecs-compute-test",
+		}},
+		// change image to nginx => httpd and then back!
+		{{
+			key:      "container_command",
+			updated:  []string{"httpd", "-D", "FOREGROUND"},
+			original: []string{"nginx", "-g", "daemon off;"},
+		}, {
+			key:      "container_image",
+			updated:  "httpd",
+			original: "nginx",
+		}},
+		// change image to nginx => httpd and then back!
+		{{
+			key:      "container_command",
+			updated:  []string{"bash", "-c", "sed -i 's/Listen 80/Listen 81/g' /usr/local/apache2/conf/httpd.conf && httpd -D FOREGROUND"},
+			original: []string{"nginx", "-g", "daemon off;"},
+		}, {
+			key:      "container_image",
+			updated:  "httpd",
+			original: "nginx",
+		}, {
+			key:      "container_port",
+			updated:  81,
+			original: 80,
+		}},
+		// change image to is_ecr_image to true
+		{{
+			key:      "container_image",
+			updated:  "public.ecr.aws/nginx/nginx:1-alpine-perl",
+			original: "nginx",
+		}, {
+			key:      "is_ecr_image",
+			updated:  true,
+			original: false,
+		}},
+	} {
+		for _, change := range changeSet {
+			mutableTFVariables[change.key] = change.updated
+		}
+		terraform.Plan(t, generateMutableTerraformOptions())
+		terraform.Apply(t, generateMutableTerraformOptions())
+		updatedECSComputeId := terraform.Output(t, terraformOptions, "ecs_compute_id")
+		updatedECSComputeAsset, updatedECSClusterAws, updatedECSServiceAws, err := getAptibleAndAWSECSServiceAndCluster(
+			t,
+			ctx,
+			c,
+			updatedECSComputeId,
+			fmt.Sprintf("%s-compute-cluster", mutableTFVariables["compute_name"].(string)),
+			mutableTFVariables["compute_name"].(string),
+		)
+		assert.Nil(t, err)
+		assertCommonValues(t, vpcId, updatedECSComputeId, vpcAsset, vpcAws, updatedECSComputeAsset, updatedECSClusterAws, updatedECSServiceAws)
```

Comment:
I don't think these common assertions are really sufficient here?

Right now we're just sending off the commands and they may or may not apply (we don't have the "wait_for_steady_state" set and we're not checking the state of it after) and just saying the fact that those indeterminate operations didn't error is good enough. I think we have to both wait for the apply to actually take place and see that it held.

Reply:
all i can do is check to see if it's active and that the service / cluster has been rolled out via the test commands and/or aws sdk (which i can do to some extent)

we have no `wait_for_steady_state` on the provider, but i can add that and increase scope but we intentionally wanted to avoid that from other discussions

Reply:
my desc on the pr was: 

> Note - i think to properly verify this, we really want to enable wait_for_steady_state

so i really think we have to do this to test

Reply:
i think, i'll add another PR for this `wait_for_steady_state` so both of us get it in our two sets of tickets + add in more elaborate testing that a task is in fact active